Assets/ScriptableObjects/Plugin Scripts/FadeoutPlugin.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[CreateAssetMenu]
+public class FadeoutPlugin : BlockPlugin
+{
+    public const float FadeSpeed = 0.1f;
+    public const float FadeMin = 0.1f;
+    public const float FadeMax = 1.0f;
+
+    // Update is called once per frame
+    public override void OnUpdate()
+    {
+	    Renderer renderer = _block.GetRenderer();
+        if (renderer.material.color.a > FadeMin && _block.IsFading())
+	        renderer.material.color = new Color(renderer.material.color.r, renderer.material.color.g, renderer.material.color.b, renderer.material.color.a - FadeSpeed);
+
+        if (renderer.material.color.a < FadeMax && !_block.IsFading())
+            renderer.material.color = new Color(renderer.material.color.r, renderer.material.color.g, renderer.material.color.b, renderer.material.color.a + FadeSpeed);
```

Comment:
We could get a reference to the camera from here. 1 Possibility would be to get a direct reference to the camera script. We may need multiple references to camera to account for different cameras across stages. That way,we inject the dependency via the inspector and not the script.

Reply:
Another thing you could do is get the camera based on tag. But could be costly on start.

Reply:
I think getting camera based on tag is a good idea. Not much different from getting blocks from tag I think.

Reply:
Ok we can try that. I'm just concerned about costs. Do note, i believe you must use _block's monobehaviour to get the object by tag.

Reply:
I think the plugin itself already has details about the block? If so, we can pass in the block object to the camera on startup